<launch>
    <arg name="global_cloud_topic" default="/global_cloud"></arg>
    <arg name="frame_id" default="world"></arg>

    <node pkg="random_map_generator" type="random_map_node" name="random_map_node" output="screen">
        <rosparam command="load" file="$(find random_map_generator)/config/rrt_corridor.yaml"/>
        <!--        <remap from="/map_generator/global_cloud" to="/global_cloud"/>-->
        <remap from="/map_generator/global_cloud" to="$(arg global_cloud_topic)"/>
        <param name="map/frame_id" value="$(arg frame_id)" type="string"></param>
    </node>

    <node pkg="random_map_generator" type="color_map_for_pointcloud" name="color_map_for_pointcloud" output="screen">
        <remap from="~cloud" to="$(arg global_cloud_topic)"></remap>
        <param name="color/frame_id" value="$(arg frame_id)" type="string"></param>
        <param name="color/min_height" value="-1" type="double"></param>
        <param name="color/max_height" value="4.5" type="double"></param>
        <param name="color/map_type_name" value="Magma" type="string"></param>
        <param name="color/publish_rate" value="1.0" type="double"></param>
        <param name="color/inverse_color" value="false" type="bool"></param>
    </node>

<!--    <node name="rvizvisualisation" pkg="rviz" type="rviz" output="screen" args="-d $(find simulator)/rviz/default.rviz" />-->

        <node pkg="rviz" type="rviz" name="test_env" args="-d $(find random_map_generator)/rviz/default.rviz"></node>

</launch>